wp/lib/bap_wp/src/precondition.ml

```diff
@@ -868,25 +868,35 @@ let _  = inline_func :=
       let post = set_fun_called post env tid in
       visit_sub env post target_sub
 
-let collect_non_null_expr (env : Env.t) (exp : Exp.t) : Constr.z3_expr list =
+(* Creates the z3_expr (not (= addr null)) *)
+let non_null_expr (env : Env.t) (addr : Exp.t) : Constr.z3_expr =
   let ctx = Env.get_context env in
+  let width = match Type.infer_exn addr with
+    | Imm n -> n
+    | Mem _ -> error "Expected %s to be an address word!%!" (Exp.to_string addr);
+      failwith "Error: in collect_non_null_expr, got a memory read instead of a word"
+    | Unk ->
+      error "Unk type: Unable to determine if %s is an address word.%!" (Exp.to_string addr);
```

Comment:
Completely different error messages between the `error` and the `failwith`. Probably not an ideal practice.

Reply:
Would it be better to combine it into a single `failwith` or `error`? Or should we just give them the same message?